src/test/java/gameframework/drawing/GameCanvasDefaultImplTest.java

```diff
@@ -0,0 +1,44 @@
+package gameframework.drawing;
+
+import static org.junit.Assert.*;
+
+import java.awt.event.KeyListener;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import gameframework.drawing.mocks.KeyListenerMock;
+
+public class GameCanvasDefaultImplTest {
+
+	protected GameCanvasDefaultImpl gameCanvas;
+	protected KeyListenerMock keyListener;
+	
+	@Before
+	public void init() {
+		gameCanvas  = new GameCanvasDefaultImpl();
+		keyListener = new KeyListenerMock();
+	}
+	
+	public KeyListener getLatestKeyListenerAdded() {
+		KeyListener[] listeners = gameCanvas.getKeyListeners();
+		return listeners.length > 0 ? listeners[listeners.length - 1] : null;
```

Comment:
if you return `null` here, you only defer the problem to the client. Please replace all that by just `gameCanvas.getKeyListeners()[listeners.length - 1]`